flake.nix

```diff
@@ -0,0 +1,16 @@
+ {
+    inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    inputs.flake-utils.url = "github:numtide/flake-utils";
+
+    outputs = { nixpkgs, flake-utils, ... }:
+      flake-utils.lib.eachDefaultSystem (system: {
+        devShells.default = nixpkgs.legacyPackages.${system}.mkShell {
+          packages = with nixpkgs.legacyPackages.${system}; [
```

Comment:
Note that I could declare Rust dependencies as well, however, I'm opting not to since: (i) I have Rust globally installed in my system (ii) this is pretty much a convenience for local development for myself mostly (iii) we can tune the flake definition in a follow-up if needed.